packages-exp/messaging-exp/src/interfaces/public-types.ts

```diff
@@ -16,37 +16,86 @@
  */
 
 /**
- * Currently supported fcm notification display parameters. Note that
- * {@link https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/notifications/NotificationOptions}
- * defines a full list of display notification parameters. This interface we only include what the
- * SEND API support for clarity.
+ * Display notification Details. They are sent through the
+ * {@link https://firebase.google.com/docs/reference/fcm/rest/v1/projects.messages#notification Send API}
+ *
+ * @public
  */
 export interface NotificationPayload {
+  /**
+   * The notification's title.
+   */
   title?: string;
+
+  /**
+   * The notification's body text.
+   */
   body?: string;
+
+  /**
+   * The URL of an image that is downloaded on the device and displayed in the notification.
+   */
   image?: string;
 }
 
 /**
+ * Options for features provided by the FCM SDK for Web. See {@link
+ * https://firebase.google.com/docs/reference/fcm/rest/v1/projects.messages#webpushfcmoptions
+ * WebpushFcmOptions}
+ *
  * @public
  */
 export interface FcmOptions {
+  /**
+   * The link to open when the user clicks on the notification.
+   */
   link?: string;
+
+  /**
+   * The label associated with the message's analytics data.
+   */
   analyticsLabel?: string;
 }
 
 /**
+ * Message payload that contains the notification payload that is represented with
+ * {@link firebase.messaging.NotificationPayload} and the data payload that contains an arbitrary
+ * number of key-value pairs sent by developers through the
+ * {@link https://firebase.google.com/docs/reference/fcm/rest/v1/projects.messages#notification Send API}
+ *
  * @public
  */
 export interface MessagePayload {
+  /**
+   * {@inheritdoc NotificationPayload}
+   */
   notification?: NotificationPayload;
+
+  /**
+   * Arbitrary key/value payload.
+   */
   data?: { [key: string]: string };
+
+  /**
+   * {@inheritdoc FcmOptions}
+   */
   fcmOptions?: FcmOptions;
+
+  /**
+   * The sender of this message.
+   */
   from: string;
+
+  /**
+   * The collapse key of the message. See more on
+   * {@link https://firebase.google.com/docs/cloud-messaging/concept-options#collapsible_and_non-collapsible_messages Non-collapsible and collapsible messages}
+   */
   collapseKey: string;
 }
 
 /**
+ * Public interface of the FirebaseMessaging SDK.
```

Comment:
Is FirebaseMessaging a literal?  If so, backtick; if not, we should probably say "Firebase Cloud Messaging."